core/rio/jsonld/src/main/java/org/eclipse/rdf4j/rio/ndjsonld/NDJSONLDParser.java

```diff
@@ -0,0 +1,97 @@
+package org.eclipse.rdf4j.rio.ndjsonld;
+
+import org.eclipse.rdf4j.model.Value;
+import org.eclipse.rdf4j.model.ValueFactory;
+import org.eclipse.rdf4j.rio.*;
+import org.eclipse.rdf4j.rio.helpers.AbstractRDFParser;
+import org.eclipse.rdf4j.rio.helpers.RDFHandlerWrapper;
+import org.eclipse.rdf4j.rio.jsonld.JSONLDParser;
+
+import java.io.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+
+public class NDJSONLDParser extends AbstractRDFParser implements RDFParser {
+
+    private JSONLDParser getJsonLDParser() {
+        JSONLDParser jsonldParserDelegate = new JSONLDParser();
+        jsonldParserDelegate.setRDFHandler(new RDFHandlerWrapper(getRDFHandler()) {
+            boolean startedRDF;
+
+            // Reuse the same parser for each line parse, but we cannot start the RDF multiple times, so call startRDF only the first time.
+            // An alternative approach would be to to create a new JSONLDParser for each row. 
+            @Override
+            public void startRDF() throws RDFHandlerException {
+                if (!startedRDF) {
```

Comment:
Instead  of this I think it's better to have an empty startRDF() method here and call rdfHandler.startRDF() in the beginning of parseInternal().
You have the same problem with endRDF() and the solution is the same: empty endRDF() here, call rdfHandler.endRDF() at the end of parseInternal().

Reply:
Entirely changed the NDJSONLDParser to convert the input to a JSON list and pass the whole model at once to the JsonLDProcessor. Done by extending the JSONLDParser